yellowbrick/classifier.py

```diff
@@ -50,6 +51,149 @@ def __init__(self, model, ax=None, **kwargs):
+
+        kwargs: keyword arguments passed to Scikit-Learn API.
+        """
+        super(ConfusionMatrix, self).fit(X, y, **kwargs)
+        if self.classes_ is None:
+            self.classes_ = self.estimator.classes_
+        return self
+
+    def score(self, X, y, sample_weight=None):
+        """
+        Generates the Scikit-Learn confusion_matrix and applies this to the appropriate axis
+
+        Parameters
+        ----------
+
+        X : ndarray or DataFrame of shape n x m
+            A matrix of n instances with m features
+
+        y : ndarray or Series of length n
+            An array or series of target or class values
+
+        sample_weight: optional, passed to the confusion_matrix
+        """
+        y_pred = self.predict(X)
+        self.confusion_matrix = confusion_matrix(y_true = y, y_pred = y_pred, labels=self.classes_, sample_weight=sample_weight)
+        #Convert confusion matrix to percent of true
+        self.confusion_matrix = np.round(self.confusion_matrix / np.sum(self.confusion_matrix, axis=1) * 100, decimals=0)
+        self.confusion_matrix = self.confusion_matrix[::-1]
+
+        return self.draw()
+
+    def draw(self):
+        # Create the axis if it doesn't exist
+        if self.ax is None:
+            self.ax = plt.gca()
+
+
+
+        #fig = plt.imshow(self.confusion_matrix, interpolation='nearest', cmap=self.cmap, vmin=0, vmax=1)
+        X = np.linspace(start=0, stop=len(self.classes_), num=len(self.classes_))
+        Y = np.linspace(start=0, stop=len(self.classes_), num=len(self.classes_))
+
+        # Draw the heatmap
+        mesh = self.ax.pcolormesh(X,Y,self.confusion_matrix, vmin=0, vmax=100,
+                             cmap=self.cmap)
+        # Add the data labels.
+        for x_index, x in np.ndenumerate(X):
+            for y_index, y in np.ndenumerate(Y):
+
+                #X and Y are one element longer than the confusion_matrix
+                if x == len(X) or y == len(Y):
+                    break
+                #np.ndenumerate returns a tuple for the index, must access first element using [0]
+                x_plus = X[(x_index[0] + 1)]
+                y_plus = Y[(y_index[0] + 1)]
+                #center the text in the middle of the block
+                text_x =  x + (x_plus - x) / 2
+                text_y =  y + (y_plus - y) / 2
+
+                #numpy doesn't like non integers as indexes
+                x_int = int(x)
+                y_int = int(y)
+
+                #make zero values more subtle
+                if self.confusion_matrix[x_int,y_int] == 0:
+                    text_color = "grey"
+                else:
+                    text_color = "black"
+
+                self.ax.text(text_y,text_x,"{:.0f}%".format(self.confusion_matrix[x_int,y_int]),va='center',ha='center', fontsize=6, color=text_color)
+
+
+        return self.ax
+
+    def finalize(self, **kwargs):
+        pass
+        #TODO add this stuff
```

Comment:
Yep, planning to. 

Note, I used the convention of WIP for "work in progress" in the title of my Pull Request. The only way to do line-by-line comments is with a pull request, so this was a way to discuss progress so far (Rebecca and I had a video chat last night) - I wasn't expecting this to be ready to merge yet.